xonsh/aliases.py

```diff
@@ -182,8 +193,19 @@ def eval_alias(
                     break
             value = value[i:]
 
+        if callable(value) and getattr(value, "return_command", False):
+            try:
+                found_return_command.append(value.__name__)
```

Comment:
Perhaps the docstring should acknowledge that it's modifying the caller's parameter before passing it through. Is there ever a situation where the caller would be surprised that this parameter was mutated?